mantidproject/mantid
# In & out of plane angle tool.

Added a tool to the instrument view that allows the user to select three crystal peaks in detector space and use these to define a plane. A fourth peak can then be selected and the in plane and out of plane angle between the peaks will be calculated.

**To test:**

 - Load some single crystal data and open the instrument view
 - Create at least 4 single crystal peaks (or any points on the instrument)
 - Choose the new alignment tool
 - Select 3 peaks to form a plane
 - Select a 4th peak to show the in-plane (theta) and out-of-plane (phi) angle of that peak

Note: the phi angle should always be calculated relative to the beam direction for the instrument i.e. `phi==0` is the beam direction.

Fixes #18120 .

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
